utils/splunk/splunk_telemetry_base.py

```diff
@@ -38,7 +66,24 @@ def check(self, instance):
         instance.update_status(current_time, self.status)
 
         try:
-            self._auth_session(instance)
+            if authentication and 'token_auth' in authentication:
+                self.log.debug("Using token based authentication mechanism")
+                persist_token_key = instance.instance_config.base_url + "token"
+                token = self.status.data.get(persist_token_key)
+                if token is None:
+                    # Since this is first time run, pick the token from conf.yaml
+                    token = authentication["token_auth"].get('initial_token')
+                if self.is_token_expired(instance, token):
+                    msg = "Current in use authentication token is expired. Please provide a valid token in the YAML " \
+                          "and restart the Agent"
+                    raise TokenExpiredException(msg)
+                if self.need_renewal(instance, token, self.initial_token_flag):
```

Comment:
will `need_renewal` return True each time the check runs because of `SplunkTelemetryBase`'s `self.initial_token_flag = True`? Better question would be; will the variable be preserved between check runs?

Reply:
I added a test for `splunk_event` and `splunk_metric` to see if the check runs 2nd time, the flag is still false.